static/hooks/useApi.jsx

```diff
@@ -0,0 +1,32 @@
+import {useCallback, useEffect, useRef} from 'react';
+
+import Client from 'app/api';
+
+/**
+ * Returns an API client that will have it's requests canceled when the owning
+ * React component is unmounted (may be disabled via options).
+ */
+function useApi({persistInFlight} = {}) {
+  const localApi = useRef();
+
+  // Lazily construct the client if we weren't provided with one
+  if (localApi.current === undefined) {
```

Comment:
A bit confusing since the hook doesn't accept a client

Reply:
ah right, this is bad copy paste